container/controller.go

```diff
@@ -287,6 +292,17 @@ func NewController(options ControllerOptions) (*Controller, error) {
 		errc <- nil
 	}()
 
+	// Load the delegate keys and auth tokens first so there's no race btwn starting the watcher routines
+	//    and making the first RPC call in getService()
+	if err := auth.LoadDelegateKeysFromFile(containerDelegateKeyFile); err != nil {
+		glog.Errorf("Unable to load delegate keys from file %s: %s", containerDelegateKeyFile, err)
+		return nil, ErrNoDelegateKey
+	}
+	if err := auth.LoadTokenFile(containerTokenFile); err != nil {
+		glog.Errorf("Unable to token file %s: %s", containerTokenFile, err)
+		return nil, ErrNoAuthToken
+	}
+
 	go auth.WatchDelegateKeyFile(containerDelegateKeyFile, keyshutdown)
```

Comment:
The first thing these do is an immediate load; was the point just to block until we have a token/key? because we have a simple way to do that already.